types/parse/index.d.ts

```diff
@@ -1138,7 +1167,16 @@ namespace Parse {
             func?: (request: BeforeFindRequest) => Promise<Query> | Promise<void> | Query | void
         ): void;
         function afterFind(arg1: any, func?: (request: AfterFindRequest) => any): void;
-        function beforeLogin(func?: (request: TriggerRequest) => any): void;
+
+        function beforeLogin(func?: (request: TriggerRequest) => Promise<void> | void): void;
```

Comment:
When a promise is expected from a user function, it's better to specify `PromiseLike` as to not cause problems if the user is using a third party promise implementation

```suggestion
        function beforeLogin(func?: (request: TriggerRequest) => PromiseLike<void> | void): void;
```